### DISCUSSION TITLE
How to change the storage location of the asset files to another server?

### DISCUSSION THREAD

**Discussion by Myles-Kadusale-Golden** [requarks/wiki#3086] [Help / Questions] [answered]
Is it possible to change the storage location of the asset files to another server?

I want this setup:
Database located in another server
Asset Files located in another server
WikiJS running in a Docker container

**Comment by mattpr** [comment, marked as answer]
You might want theme/logo files to be served from another server, but assets in general probably not.  
The reason why is that wiki.js is providing access control rules based on paths.

This is why if you upload a logo ("asset") to display in wiki.js (say `/logos/mylogo.svg`) you get a 403 for that image on the loginpage unless you grant read permission for Guest (not logged in) users for path prefix `/logos/`

Well for themes/logos...that's probably fine as you are probably fine with them being public.  They could probably be treated separately from other uploaded wiki assets to avoid permission confusion anyway.  

But say you have a pdf of your Q4 financials uploaded as attachment to wiki page.  That is an asset.  You want the wiki protecting that asset with access control.  If you just serve those files from a different webserver, how is the access control handled?

Does a user have permission to upload a new asset and link to it in a wiki page?  wiki.js handles the upload/control over that... so post-upload wiki.js would need to pass the uploaded asset off to be served by another service in the future.  Gets complex fast.

If you want to do static serving for assets anyway, just run nginx in front of wiki.js and add specific location blocks to the nginx config for asset paths so nginx intercepts and handles these requests instead of passing them to the wiki.js upstream.  You can even have another nginx server as an asset upstream.  In this scenario you would still have all requests coming through a single nginx (unless you have load balancing in front of it).  If you wanted to segregate it on the client side then you would need different request domains for assets than for wiki content and this would need to be supported via wiki.js since the asset references are generated by the wiki.js app.  

But as I said, other than perhaps theme and public logos/images (think login page assets) you probably don't want to do this for access control reasons.  Even if your wiki is fully public (many public readers, few logged-in editors) you probably don't want this because as an editor you want to have control over what you make public when and not have everything public as soon as it is uploaded.

**Reply by Myles-Kadusale-Golden** [reply]
You are right.  I found also that the files are stored in the database
